airflow/gcp/hooks/gsheets.py

```diff
@@ -0,0 +1,393 @@
+
+
+class GSheetsHook(GoogleCloudBaseHook):
+    """
+    Interact with Google Sheets via GCP connection
+    Reading and writing cells in Google Sheet:
+    https://developers.google.com/sheets/api/guides/values
+
+    :param gcp_conn_id: The connection ID to use when fetching connection info.
+    :type gcp_conn_id: str
+    :param spreadsheet_id: The Google Sheet ID to interact with
+    :type spreadsheet_id: str
+    :param api_version: API Version
+    :type api_version: str
+    :param delegate_to: The account to impersonate, if any.
+        For this to work, the service account making the request must have
+        domain-wide delegation enabled.
+    :type delegate_to: str
+    """
+
+    def __init__(
+        self,
+        gcp_conn_id: str = 'google_cloud_default',
+        spreadsheet_id: str = None,
+        api_version: str = 'v4',
+        delegate_to: str = None
+    ) -> None:
+        super().__init__(gcp_conn_id, delegate_to)
+        self.gcp_conn_id = gcp_conn_id
+
+        if not spreadsheet_id:
+            raise AirflowException("The spreadsheet_id must be passed!")
+
+        self.spreadsheet_id = spreadsheet_id
+        self.api_version = api_version
+        self.delegate_to = delegate_to
+        self.num_retries = self._get_field('num_retries', 5)
+        self._conn = None
+
+    def get_conn(self):
+        """
+        Retrieves connection to Google Sheets.
+        :return: Google Sheets services object.
+        :rtype: dict
+        """
+        if not self._conn:
+            http_authorized = self._authorize()
+            self._conn = build('sheets', self.api_version, http=http_authorized, cache_discovery=False)
+
+        return self._conn
+
+    @GoogleCloudBaseHook.catch_http_exception
+    def get_values(
+        self,
+        range_: str,
+        major_dimension: str = 'DIMENSION_UNSPECIFIED',
+        value_render_option: str = 'FORMATTED_VALUE',
+        date_time_render_option: str = 'SERIAL_NUMBER'
+    ) -> dict:
+        """
+        Gets values from Google Sheet from a single range
+        https://developers.google.com/sheets/api/reference/rest/v4/spreadsheets.values/get
+
+        :param range_: The A1 notation of the values to retrieve.
+        :type range_: str
+        :param major_dimension: Indicates which dimension an operation should apply to.
+            DIMENSION_UNSPECIFIED, ROWS, or COLUMNS
+        :type major_dimension: str
+        :param value_render_option: Determines how values should be rendered in the output.
+            FORMATTED_VALUE, UNFORMATTED_VALUE, or FORMULA
+        :type value_render_option: str
+        :param date_time_render_option: Determines how dates should be rendered in the output.
+            SERIAL_NUMBER or FORMATTED_STRING
+        :type date_time_render_option: str
+        :return: Google Sheets API response.
+        :rtype: dict
+        """
+        if not range_:
+            raise AirflowException("The 'range_' argument must be passed!")
```

Comment:
This assertion is redundant because `range_` must be passed as it has no default value.